pandas/types/cast.py

```diff
@@ -265,6 +266,9 @@ def _maybe_promote(dtype, fill_value=np.nan):
     elif is_datetimetz(dtype):
         if isnull(fill_value):
             fill_value = iNaT
+    elif is_unsigned_integer_dtype(dtype):
+        if isnull(fill_value):
```

Comment:
@gfyoung rather insiduous issue with uint64 and take, so missing values were being promoted to ``nan``. which casuses a float conversion. So we need some more testing around this :>